src/main/java/org/candlepin/subscriptions/resource/CapacityResource.java

```diff
@@ -179,10 +185,18 @@ public CapacityReportByMetricId getCapacityReportByMetricId(
       ServiceLevelType sla,
       UsageType usage) {
 
+    ProductId productId;
+    MetricId metricId;
+    try {
+      productId = ProductId.fromString(productIdValue);
+      metricId = MetricId.fromString(metricIdValue);
```

Comment:
And we might as well do a `MetricId` validator while we're at it.

Reply:
Nacking for above reason